beater/authorization/builder.go

```diff
@@ -46,18 +46,23 @@ const (
 )
 
 // NewBuilder creates authorization builder based off of the given information
-func NewBuilder(cfg *config.Config) (*Builder, error) {
+// if apm-server.api_key is enabled, authorization is granted/denied solely
+// based on the request Authorization header
+func NewBuilder(cfg config.Config) (*Builder, error) {
 	b := Builder{}
 	b.fallback = AllowAuth{}
 	if cfg.APIKeyConfig.IsEnabled() {
+		// do not use username+password for API Key requests
+		cfg.APIKeyConfig.ESConfig.Username = ""
+		cfg.APIKeyConfig.ESConfig.Password = ""
```

Comment:
We inherit `output.elasticsearch.api_key` from libbeat. Do you mind also removing this field here, as it would lead to the same issues as username+password. 
(No sense in me creating a dedicated PR from master for that, as you are moving where this information is removed in this PR).